odl/tomo/geometry/conebeam.py

```diff
@@ -187,10 +206,17 @@ def __init__(self, apart, dpart, src_radius, det_radius,
         self.__src_to_det_init = src_to_det_init
         # `check_bounds` is needed for both detector and geometry
         check_bounds = kwargs.get('check_bounds', True)
-        detector = Flat1dDetector(dpart, axis=det_axis_init,
-                                  check_bounds=check_bounds)
+        if det_curve_radius is None:
+            detector = Flat1dDetector(dpart, axis=det_axis_init,
+                                      check_bounds=check_bounds)
+        else:
+            det_curve_radius = float(det_curve_radius)
```

Comment:
This will be done by `CircularDetector`, so it's okay to just pass on the parameter.